src/Mario.purs

```diff
@@ -71,8 +74,8 @@ walk _ _ c = applyFriction c
   applyFriction :: Character -> Character
   applyFriction c | c.dx == 0.0 = c
   applyFriction c | abs c.dx <= friction c = c { dx = 0.0 }
-  applyFriction c | c.dx > 0.0 = c { dx = c.dx - friction c }
-  applyFriction c | c.dx < 0.0 = c { dx = c.dx + friction c }
+  applyFriction c | abs c.dx >= friction c = c { dx = 0.0 }
```

Comment:
No, this won't work. Only the last case needed to be changed, as in my earlier comment. I should add comments to each case of this function.